beamer/events.py

```diff
@@ -207,5 +214,13 @@ def fetch(self) -> list[Event]:
                     from_block = BlockNumber(to_block + 1)
 
             self._next_block_number = from_block
+
+            result.append(
```

Comment:
thanks for the change. couldn't we move this directly into line 204?

Reply:
I'd rather do the update of the latest block after all events that happened before are processed. I don't see any problem right now, but think that's the more sensible approach.

Reply:
Merging for now, if you disagree we can fix this later.